File: sorting/c/bubble_sort.c
#include <stdio.h>
#include <stdlib.h>

void bubble_sort(int* array, const int n)
{
	int i, swapped;
	do
	{
		swapped = 0;
		for(i = 0; i < n - 1; i++)
			if(array[i] > array[i + 1])
			{
				int tmp = array[i];
				array[i] = array[i + 1];
				array[i + 1] = tmp;

				swapped = 1;
			}
	} while( swapped );
}

File: sorting/java/BubbleSort.java

public class BubbleSort implements SortAlgorithm
{
	public void sort(int[] array)
	{
		boolean swapped = false;
		do
		{
			swapped = false;
			for(int i = 1; i < array.length; i++)
				if(array[i - 1] > array[i])
				{
					int tmp = array[i - 1];
					array[i - 1] = array[i];
					array[i] = tmp;
					swapped = true;
				}
		} while(swapped);
	}
}
